File: easy_22_capitalize_first_letter_of_each_word/solution.py
"""Reference implementation for the platform's official solution.

Update this file with the fully working algorithm that matches the
`description.txt` contract and is compatible with the wrappers.
"""

from __future__ import annotations

import sys
from typing import TextIO


class Solution:
    def capitalizeWords(self, sentence):
        """
        :type sentence: str
        :rtype: str
        """
        words = sentence.split()
        capitalized_words = []
        for word in words:
            if word:  # Check if word is not empty
                capitalized_word = word[0].upper() + word[1:].lower() if len(word) > 1 else word.upper()
                capitalized_words.append(capitalized_word)
        return ' '.join(capitalized_words)


def solve(infile: TextIO = sys.stdin, outfile: TextIO = sys.stdout) -> None:
    """Read from ``infile`` and write the answer to ``outfile``.

    This signature enables deterministic unit testing while still working with
    Judge0 (which provides stdin/stdout). Replace the body with the final
    algorithm for the problem, including any necessary parsing.
    """

    sentence = infile.read().strip()
    
    # Create solution instance and call the method
    solution = Solution()
    result = solution.capitalizeWords(sentence)
    
    # Output the result
    outfile.write(f"{result}\n")


if __name__ == "__main__":
    solve()

File: easy_22_capitalize_first_letter_of_each_word/solution.cpp
#include <algorithm>
#include <iostream>
#include <iterator>
#include <sstream>
#include <string>
#include <vector>
#include <cctype>

using namespace std;

class Solution {
public:
    string capitalizeWords(const string& sentence) {
        istringstream iss(sentence);
        string word;
        vector<string> words;
        
        while (iss >> word) {
            if (!word.empty()) {
                word[0] = toupper(word[0]);
                for (size_t i = 1; i < word.length(); i++) {
                    word[i] = tolower(word[i]);
                }
                words.push_back(word);
            }
        }
        
        string result;
        for (size_t i = 0; i < words.size(); i++) {
            result += words[i];
            if (i < words.size() - 1) result += " ";
        }
        
        return result;
    }
};

void solve(istream& in = cin, ostream& out = cout) {
    // Read input
    string sentence;
    getline(in, sentence);
    
    // Create solution instance and call the method
    Solution solution;
    string result = solution.capitalizeWords(sentence);
    
    // Output the result
    out << result << endl;
}

int main() {
    solve();
    return 0;
}
